src/components/app/ProjectPreview.svelte

```diff
@@ -85,9 +145,43 @@
                 : value
                   ? value.getRepresentativeText($locales)
                   : EXCEPTION_SYMBOL,
+            isCustomCharacter: false,
+            characterName: null,
         };
     });
 
+    // Add a state variable to hold the character
+    let character = $state<Character | null>(null);
+
+    // Effect to fetch the character from the database when characterName changes
+    $effect(() => {
+        // Clear character if characterName is not valid
+        if (!isCustomCharacter || !characterName) {
+            character = null;
+            return;
+        }
+
+        // Make sure DB.Characters exists and has getByName method
+        if (!DB.Characters || typeof DB.Characters.getByName !== 'function') {
+            console.error('DB.Characters or getByName method not available');
+            return;
+        }
+
+        // Fetch the character
+        DB.Characters.getByName(characterName)
+            .then((result) => {
+                if (result) {
+                    character = result;
+                } else {
+                    character = null;
+                }
+            })
+            .catch((error) => {
+                console.error('bug', error);
```

Comment:
This is not a suitable error message. If there was an error loading the character, there should be a more descriptive console message explaining the context of what was happening (e.g., the function already returns null or undefined for different not found conditions, so this is a more fatal error, like the connection to Firebase being broken, or there being a permissions error).